### ISSUE TITLE
Steam overlay lag & usability

### ISSUE THREAD

**Issue by CCC200** [Gamua/Adobe-Runtime-Support#505]
Hello,

In the newest version of the AIR Runtime (33.1.1.259 as of writing) the overlay that hooks into applications when running them through Steam is nearly unusable. It lags heavily, freezes, & locks up frequently. It works fine in non-AIR applications (even old flash applications seem to be fine), and I suspect this has something to do with the AIR renderer.

In Direct mode, the overlay can render, but is extremely laggy as I just described. In CPU mode, it refuses to render at all. I tried manually setting GPU mode by editing the config after building, and found that while still very frustrating and slow, it wouldn't freeze up anymore; but this is hardly a useful hack. Our application relies on this overlay to make in-game purchases, and having our best option be a config hack that makes it _just_ technically usable is too much for most customers to put up with.

I've tried my best to mess with compiler options but can't seem to create an outcome better than the GPU mode.

EDIT: To clarify this is using Animate, not Starling. Not sure if Starling's better GPU use solves this issue.

**Comment by CCC200** [comment]
> It marked as closed yet no solution has given, has anyone found a solution for it?

I closed it as I doubt there's any chance they work on this. I believe it has to do with how the stage refreshes when there aren't active moving elements in frame, which is a pretty deep issue. I did end up recently finding a workaround however!

[Ventero/FRESteamWorks](https://github.com/Ventero/FRESteamWorks) has the event `RESPONSE_OnGameOverlayActivated` that is fired every time the overlay opens/closes. What I basically did was this:

1. Create a movieclip of a square that rotates every frame
2. When the overlay is opened, spawn the square in the corner of the screen (it has to be visible but I don't think size matters) and loop the rotation animation
3. When the overlay is closed, despawn the square

This forces the screen to refresh every frame, and the overlay ended up running a lot better! Additionally, if for whatever reason your application doesn't use a full 16:9 resolution in fullscreen, fill the area outside your stage. Steam notification popups always appear in the corner and this seems to prevent issues with those.

Hope this helps!
